connector_dns/unit/binder.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Elico Corp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from datetime import datetime
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from openerp.addons.connector.connector import Binder
+from ..backend import dns
+
+
+class DNSPodBinder(Binder):
+    """ Generic Binder for DNSPod """
+
+
+@dns
+class DNSPodModelBinder(DNSPodBinder):
+    """
+    Bindings are done directly on the binding model.
+
+    Binding models are models called ``dns.{normal_model}``,
+    like ``dns.record`` or ``dns.domain``.
+    They are ``_inherits`` of the normal models and contains
+    the DNS ID, the ID of the DNS Backend and the additional
+    fields belonging to the DNS instance.
+    """
+    _model_name = [
+        'dns.record',
+        'dns.domain'
+    ]
+
+    def to_openerp(self, external_id, unwrap=False):
+        """ Give the OpenERP ID for an external ID
+
+        :param external_id: external ID for which we want the OpenERP ID
+        :param unwrap: if True, returns the openerp_id of the dns_xx record,
+                       else return the id (binding id) of that record
+        :return: a record ID, depending on the value of unwrap,
+                 or None if the external_id is not mapped
+        :rtype: int
+        """
+        binding_ids = self.session.search(
+            self.model._name,
+            [('dns_id', '=', str(external_id)),
+             ('backend_id', '=', self.backend_record.id)])
+        if not binding_ids:
+            return None
+        assert len(binding_ids) == 1, "Several records found: %s" % binding_ids
+        binding_id = binding_ids[0]
+        if unwrap:
+            return self.session.read(self.model._name,
+                                     binding_id,
+                                     ['openerp_id'])['openerp_id'][0]
```

Comment:
Here's some pesky [core logic break I was referencing](https://github.com/OCA/infrastructure-dns/pull/4/files#r76639007). Again, recommend `odoo_id` instead of `openerp_id` but we do need something.